src/cloud-providers/gcp/provider.go

```diff
@@ -76,6 +76,29 @@ func getIPs(instance *computepb.Instance) ([]netip.Addr, error) {
 	return podNodeIPs, nil
 }
 
+func (p *gcpProvider) getImageSizeGB(ctx context.Context, image string) (int64, error) {
```

Comment:
May be you can use rootVolSize to align with other code - https://github.com/confidential-containers/cloud-api-adaptor/blob/68df7a40ea5906884b3f9b34252a26fb00af2056/src/cloud-providers/aws/provider.go#L100
Also can the size be changed based on input value during VM creation? If yes, then you can follow a similar approach like existing AWS code.

Reply:
@bpradipt completely missed that logic. I just force-pushed with a small change, that I believe is doing the same. I tested and it worked.